perfkitbenchmarker/timing_util.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2014 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Utilities for generating timing samples."""
+
+from contextlib import contextmanager
+import time
+
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import flags_validators
+from perfkitbenchmarker import sample
+
+
+# Name of the measurements configuration flag as a user would type it.
+MEASUREMENTS_FLAG_NAME = 'timing_measurements'
+# Valid options that can be included in the flag's list value.
+MEASUREMENTS_NONE = 'NONE'
+MEASUREMENTS_END_TO_END_RUNTIME = 'END_TO_END_RUNTIME'
+MEASUREMENTS_RUNTIMES = 'RUNTIMES'
+MEASUREMENTS_TIMESTAMPS = 'TIMESTAMPS'
+MEASUREMENTS_ALL = (
+    MEASUREMENTS_NONE, MEASUREMENTS_END_TO_END_RUNTIME, MEASUREMENTS_RUNTIMES,
+    MEASUREMENTS_TIMESTAMPS)
+
+
+def ValidateMeasurementsFlag(options_list):
+  """Verifies correct usage of the measurements configuration flag.
+
+  The user of the flag must provide at least one option. All provided options
+  must be valid. The NONE option cannot be combined with other options.
```

Comment:
If none were the default value, this would be fine, because the flag could be omitted, but to match previous behavior, the default is END_TO_END_RUNTIME. As a result, the user would need to specify --timing_measurements='' or similar to mark an empty value, and that would be more confusing than --timing-measurements=none.